pageserver/src/layered_repository.rs

```diff
@@ -1623,6 +1628,22 @@ impl LayeredTimeline {
                 );
                 Ok(ZERO_PAGE.clone())
             } else {
+                if let RelishTag::Relation(rel_tag) = &rel {
+                    let page_cache = PAGE_CACHE.read().unwrap();
+                    if let Some((img_lsn, img)) = page_cache.lookup(
+                        self.tenantid,
+                        self.timelineid,
+                        *rel_tag,
+                        blknum,
+                        request_lsn,
+                    ) {
+                        let (last_wal_lsn, _rec) = data.records.last().unwrap();
+                        if last_wal_lsn <= &img_lsn {
+                            return Ok(Bytes::from(img.to_owned()));
+                        }
+                    }
+                }
```

Comment:
We could also use it to trim the amount of WAL needed to replay

Reply:
Yeah. It's also a bit sad that we look up the cache only after going through all the work to collect the WAL records from the layers. We should check the cache first.